src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/tests/latest/test_consumption_commands.py

```diff
@@ -165,3 +188,41 @@ def test_list_reservations_details_with_reservationid(self):
         reservations_details_list = self.cmd('consumption reservations details list -r ca69259e-bd4f-45c3-bf28-3f353f9cce9b -i f37f4b70-52ba-4344-a8bd-28abfd21d640 -s ''2017-12-01'' -e ''2017-12-07''').get_output_in_json()
         self.assertTrue(reservations_details_list)
         self._validate_reservation_details(reservations_details_list[0])
+
+    def test_consumption_marketplace_list(self):
+        marketplace_list = self.cmd('consumption marketplace list').get_output_in_json()
+        self.assertTrue(marketplace_list)
+        all(self._validate_marketplace(marketplace_item) for marketplace_item in marketplace_list)
+
+    def test_consumption_marketplace_list_billing_period_filter(self):
+        marketplace_list = self.cmd('consumption marketplace list -p 201804-1 -t 1').get_output_in_json()
+        self.assertTrue(marketplace_list)
+        self.assertTrue(len(marketplace_list) == 1)
+        all(self._validate_marketplace(marketplace_item) for marketplace_item in marketplace_list)
+
+    def test_consumption_marketplace_list_billing_period(self):
+        marketplace_list = self.cmd('consumption marketplace list -p 201804-1').get_output_in_json()
+        self.assertTrue(marketplace_list)
+        all(self._validate_marketplace(marketplace_item, '201804-1') for marketplace_item in marketplace_list)
+
+    def test_consumption_budget_usage_create(self):
+        output_budget = self.cmd('az consumption budget create -b usagetypebudget1 -a 20 -s 2018-02-01 -e 2018-10-01 -t annually -c usage -m 0dfadad2-6e4f-4078-85e1-90c230d4d482').get_output_in_json()
+        self.assertTrue(output_budget)
+        self._validate_budget(output_budget)
+
+    def test_consumption_budget_create(self):
+        output_budget = self.cmd('consumption budget create -b ''costbudget'' -c ''cost'' -a 100.0 -s ''2018-02-01'' -e ''2018-10-01'' -t ''monthly''').get_output_in_json()
```

Comment:
Generally speaking, you don't need to double single quotes around things.

Reply:
done